memory.go

```diff
@@ -22,8 +24,10 @@ func CreateInMemory() *InMemory {
 
 // Store stores a message and returns its storage ID
 func (memory *InMemory) Store(m *data.Message) (string, error) {
+	memory.mu.Lock()
 	memory.Messages = append(memory.Messages, m)
 	memory.MessageIDIndex[string(m.ID)] = len(memory.Messages) - 1
+	memory.mu.Unlock()
```

Comment:
this should be a `defer memory.mu.Unlock()` after line 27 instead, e.g.
```go
memory.mu.Lock()
defer memory.mu.Unlock()
```

although unlikely, if either line 28 or 29 panics, the mutex would still be locked and would cause all other goroutines to block (see https://gist.github.com/ian-kent/06efa5b60d3b9f54ee07648e770a1958 for an example)

the same applies to the other mutex unlocks in the rest of `memory.go`